src/state_diff.rs

```diff
@@ -980,13 +951,15 @@ mod tests {
 
         let mut trie1 = collection.trie_for(crate::empty_trie_hash());
         for (key, value) in &keys1 {
+            #[allow(clippy::explicit_auto_deref)]
             trie1.insert(key, *value);
         }
         let patch = trie1.into_patch();
         let first_root = collection.apply_increase(patch, crate::gc::tests::no_childs);
 
         let mut trie2 = collection.trie_for(crate::empty_trie_hash());
         for (key, value) in &keys2 {
+            #[allow(clippy::explicit_auto_deref)]
```

Comment:
there's a bug in clippy: compiler is not able to auto-deref , what clippy says it's capable to